plugin/pulpcore/plugin/changeset/model.py

```diff
@@ -158,30 +203,85 @@ class RemoteArtifact(Remote):
         >>> model = Artifact(...)  # DB model instance.
         >>> download = ...
         >>> ...
-        >>> artifact = RemoteArtifact(model, download)
+        >>> artifact = PendingArtifact(model, 'http://lion.rpm', 'lion.rpm')
         >>>
     """
 
     __slots__ = (
+        'url',
+        'relative_path',
+        'monitor',
         'content',
-        'download',
-        'path'
+        '_path',
     )
 
-    def __init__(self, model, download):
+    def __init__(self, model, url, relative_path):
         """
-
         Args:
-            model (pulpcore.plugin.models.Artifact): An artifact model instance.
+            model (pulpcore.plugin.models.Artifact): A pending artifact model instance.
                 This instance will be used to store a newly created or updated
-                artifact in the DB.
-            download (pulpcore.download.Download): A An object used to download the content.
+                pending artifact in the DB.
+            url (str): The URL used to download the artifact.
+            relative_path (str): The relative path within the content.
         """
-        super(RemoteArtifact, self).__init__(model)
-        self.download = download
-        self.download.attachment = self
-        self.path = download.writer.path
+        super().__init__(model)
+        self.url = url
+        self.relative_path = relative_path
+        self.monitor = None
         self.content = None
+        self._path = ''
+
+    @property
+    def changeset(self):
+        """
+        Returns:
+            pulpcore.plugin.changeset.Changeset: The active changeset.
+        """
+        return self.content.changeset
+
+    @property
+    def importer(self):
+        """
+        Returns:
+            pulpcore.plugin.models.Importer: An importer.
+        """
+        return self.changeset.importer
+
+    def downloader(self):
+        """
+        Get a downloader.
+
+        Returns:
+            pulpcore.download.Download: A downloader.
+        """
+        def succeeded(event):
+            self._path = event.download.writer.path
+        download = self.importer.get_download(
+            self.url,
+            self.relative_path,
+            self.model)
+        download.attachment = self
+        download.register(Event.SUCCEEDED, succeeded)
+        self.monitor = DownloadMonitor(download)
+        return download
+
+    def artifact_q(self, artifact=None):
+        """
+        Get a query for the actual artifact.
+
+        Args:
+            artifact: A artifact model.
```

Comment:
It would be good to document that this is optional, and what happens when it's not provided.